pkg/retrieval/retrieval_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package retrieval_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p/mock"
+	"github.com/ethersphere/bee/pkg/p2p/protobuf"
+	"github.com/ethersphere/bee/pkg/retrieval"
+	storemock "github.com/ethersphere/bee/pkg/storage/mock"
+)
+
+type mockPeerSuggester struct {
+	returnValue string
+}
+
+func (v mockPeerSuggester) SuggestPeer(_ []byte) (string, error) {
+	return v.returnValue, nil
+}
+
+// TestDelivery tests that a naive request -> delivery flow works
+func TestDelivery(t *testing.T) {
+	logger := logging.New(ioutil.Discard)
+
+	mockStorer := storemock.NewStorer()
+	reqAddr := []byte("reqaddr")
+	reqData := []byte("data data data")
+
+	// put testdata in the mock store of the server
+	_ = mockStorer.Put(context.TODO(), reqAddr, reqData)
+
+	// create the server that will handle the request and will serve the response
+	server := retrieval.New(retrieval.Options{
+		Storer: mockStorer,
+		Logger: logger,
+	})
+	recorder := mock.NewRecorder(
+		mock.WithProtocols(server.Protocol()),
+	)
+
+	// client mock storer does not store any data at this point
+	// but should be checked at at the end of the test for the
+	// presence of the reqAddr key and value to ensure delivery
+	// was successful
+	clientMockStorer := storemock.NewStorer()
+
+	ps := mockPeerSuggester{returnValue: "/p2p/QmZt98UimwpW9ptJumKTq7B7t3FzNfyoWVNGcd8PFCd7XS"}
```

Comment:
This return value looks like an underlay address. We should deal only with overlay addresses for all protocols. Even it is just a string, it should represent a random set of characters, until we user swarm.Address (recently added) for overlay address.